### PR TITLE
Resolve broken links and other minor issues in Chapter 6.

### LINKED ISSUE #207: 6. Resolve broken links and other minor issues

### PR DESCRIPTION
**Pull request by kirtchev-adacore**
Fixes #207

### CODE REVIEW

File: src/expressions.rst [hunk 1]
```diff
@@ -323,21 +317,22 @@ context`.
 :dp:`fls_od0h3v40kjp6`
 An invocation of the ``addr_of!()`` :t:`macro` expands to a :t:`constant
 expression` allowed in any :t:`constant context` and :t:`constant function`,
-subject to the same restrictions as a :t:`shared borrow expression`.
+subject to the same restrictions as a :t:`mutable borrow expression`.
```
[the remark this answers is not in the document]
Comment (kirtchev-adacore) [reply src/expressions.rst:320]: This was a mistake, should be "immutable".

File: src/expressions.rst [hunk 2]
```diff
@@ -3505,6 +3506,9 @@ Loop Expressions
    LoopExpression ::=
        Label? LoopContent
 
+   Label ::=
+       $$'$$ NonKeywordIdentifier
```
[the remark this answers is not in the document]
Comment (kirtchev-adacore) [reply src/expressions.rst:3510]: Good catch!

File: src/glossary.rst [hunk 3]
```diff
@@ -1204,6 +1254,14 @@ restarts a :t:`loop expression`.
 :dp:`fls_smwcz2xw9o1f`
 See :s:`ContinueExpression`.
 
+.. _fls_nC4Knv4tpenW:
+
+control flow boundary
+^^^^^^^^^^^^^^^^^^^^^
+
+:dp:`fls_SmipZJDp02ij`
+A :dt:`control flow boundary` is a ??? HELP LUKAS!
```
[the remark this answers is not in the document]
Comment (Veykril) [reply src/glossary.rst]: Defined this in d3b9e13

**Review by Veykril — APPROVED** [review]

**Comment by Veykril** [comment]
bors merge

**Pull request merged**